lib/modules/platform/github/index.ts

```diff
@@ -553,12 +553,12 @@ export async function initRepo({
     // Base branch may be configured but defaultBranch is always fixed
     logger.debug(`${repository} default branch = ${config.defaultBranch}`);
     // GitHub allows administrators to block certain types of merge, so we need to check it
-    if (repo.rebaseMergeAllowed) {
-      config.mergeMethod = 'rebase';
-    } else if (repo.squashMergeAllowed) {
+    if (repo.squashMergeAllowed) {
```

Comment:
we should support user configured merge method in same PR, like we do on other platforms. so we unify logic

Reply:
I just want to confirm that you are asking me to implement this: https://github.com/renovatebot/renovate/issues/16884

Not against it at all, just don't want to create more noise for you guys if you meant something else.

Reply:
I think it should be done separately / after